packages/lesswrong/lib/editor/make_editable.ts

```diff
@@ -161,69 +270,25 @@ export const makeEditable = <N extends CollectionNameString>({
-        type: 'Revision',
-        arguments: 'version: String',
-        resolver: async (doc: ObjectsByCollectionName[N], args: {version?: string}, context: ResolverContext): Promise<DbRevision|null> => {
-          const { version } = args;
-          const { currentUser, Revisions } = context;
-          const field = fieldName || "contents"
-          const { checkAccess } = Revisions
-          if (version) {
-            if (version === "draft") {
-              // If version is the special string "draft", that means
-              // instead of returning the latest non-draft version
-              // (what we'd normally do), we instead return the latest
-              // version period, including draft versions.
-              const revision = await Revisions.findOne({documentId: doc._id, fieldName: field}, {sort: {editedAt: -1}})
-
-              if (!revision) return null;
-              return await checkAccess(currentUser, revision, context) ? revision : null
-            } else {
-              const revision = await Revisions.findOne({documentId: doc._id, version, fieldName: field})
-              if (!revision) return null;
-              return await checkAccess(currentUser, revision, context) ? revision : null
-            }
-          }
-          const docField = (doc as AnyBecauseTodo)[field];
-          if (!docField) return null
-
-          const result: DbRevision = {
-            ...docField,
-            // we're specifying these fields manually because docField doesn't have them, 
-            // or because we need to control the permissions on them.
-            //
-            // The reason we need to return documentId and collectionName is because this 
-            // entire result gets recursively resolved by revision field resolvers, and those
-            // resolvers depend on these fields existing.
-            _id: `${doc._id}_${fieldName}`, //HACK
-            documentId: doc._id,
-            collectionName: collection.collectionName,
-            editedAt: new Date(docField?.editedAt ?? Date.now()),
-            originalContents: getOriginalContents(context.currentUser, doc, docField.originalContents),
-          }
-          return result
-          //HACK: Pretend that this denormalized field is a DbRevision (even though it's missing an _id and some other fields)
-        }
-      },
+      resolveAs: buildEditableResolver(collection, fieldName, normalized),
       form: {
         label,
-        hintText: hintText,
-        fieldName: fieldName || "contents",
+        hintText,
+        fieldName,
         collectionName,
         commentEditor,
         commentStyles,
         hideControls,
       },
     },
 
-    [`${fieldName || "contents"}_latest`]: {
+    [`${fieldName}_latest`]: {
       type: String,
       canRead: ['guests'],
       optional: true,
     },
 
-    [camelCaseify(`${fieldName}Revisions`)]: {
+    [fieldName === "contents" ? "revisions" : camelCaseify(`${fieldName}Revisions`)]: {
```

Comment:
This doesn't look backwards-compatible in the `contents` case, but I'm not 100% sure that this field was ever used anywhere (i.e. I don't see any instances of `contentsRevisions` anywhere in the codebase).  Might be misunderstanding something?

Same thing for the version field below

Reply:
For `contents`, `fieldName` is `undefined`, and `camelCaseify` lowercases the first letter, so `camelCaseify(`${fieldName}Revisions`)` becomes "revisions" (this PR changes it so that `fieldName` is never `undefined` though, hence this change). If the change wasn't backwards compatible it would also be visible after `yarn generate` (which is how I saw that I actually needed to make this change).